test/qtismtest/runtime/expressions/operators/PatternMatchProcessorTest.php

```diff
@@ -111,6 +111,8 @@ public function patternMatchProvider()
             [new QtiString('stringString'), '.*', true], // in xml schema 2, dot matches white-spaces
             [new QtiString('^String$'), 'String', false], // No carret nor dollar in xml schema 2
             [new QtiString('^String$'), '^String$', true],
+            [new QtiString('^String$'), '[^String]*$', false],
+            [new QtiString('aaa$'), '[^String]*$', true],
```

Comment:
Unnecessary $ in this two tests, it steals the focus on the caret.
```suggestion
            [new QtiString('String'), '[^String]*', false],
            [new QtiString('aaa'), '[^String]*', true],
```
            [new QtiString('aaa$'), '[^String]*$', true],